src/collapse/common.rs

```diff
@@ -0,0 +1,794 @@
+use std::borrow::Cow;
+use std::io;
+use std::mem;
+use std::sync::Arc;
+
+use chashmap::CHashMap;
+use crossbeam::channel;
+use fnv::FnvHashMap;
+use lazy_static::lazy_static;
+
+const CAPACITY_HASHMAP: usize = 512;
+pub(crate) const CAPACITY_READER: usize = 128 * 1024;
+pub(crate) const DEFAULT_NSTACKS_PER_JOB: usize = 100;
+const NBYTES_PER_STACK_GUESS: usize = 1024;
+const RUST_HASH_LENGTH: usize = 17;
+
+lazy_static! {
+    #[doc(hidden)]
+    pub static ref DEFAULT_NTHREADS: usize = num_cpus::get();
+}
+
+/// Private trait for internal library authors.
+///
+/// If you implement this trait, your type will implement the public-facing
+/// `Collapse` trait as well. Implementing this trait gives you parallelism
+/// for free as long as you adhere to the requirements described in the
+/// comments below.
+pub trait CollapsePrivate: Clone + Send + Sized + Sized {
+    // *********************************************************** //
+    // ********************* REQUIRED METHODS ******************** //
+    // *********************************************************** //
+
+    /// Some formats, such as `dtrace`, contain a header or other non-stack
```

Comment:
Usually, the first line should be a brief description of the method. I'd propose leading with:
```rust
/// Process any header lines that precede the main body of samples.
///
```